hardware_interface/src/resource_manager.cpp

```diff
@@ -608,46 +608,71 @@ class ResourceStorage
   template <class HardwareT>
   void import_state_interfaces(HardwareT & hardware)
   {
-    try
+    std::vector<std::string> interface_names;
+    std::vector<StateInterface::SharedPtr> interfaces = hardware.export_state_interfaces();
+
+    interface_names.reserve(interfaces.size());
+    for (auto const & interface : interfaces)
     {
-      auto interfaces = hardware.export_state_interfaces();
-      std::vector<std::string> interface_names;
-      interface_names.reserve(interfaces.size());
-      for (auto & interface : interfaces)
+      try
+      {
+        interface_names.push_back(add_state_interface(interface));
+      }
```

Comment:
I think we don't have to itterate here, but can just use:

```
add_state_interfaces(interfaces);
```

Then we can remove the `add_state_interface` method, to peel off one layer :smile:

Reply:
done, boss!